### DISCUSSION TITLE
StackLayout with TouchEffect inside ScrollView

### DISCUSSION THREAD

**Discussion by AlleSchonWeg** [xamarin/XamarinCommunityToolkit#686] [Q&A] [answered]
Hi @AndreiMisiukevich,
i have the following simplefied xaml:

```
<ScrollView VerticalOptions="FillAndExpand"
						Margin="0,0,0,0">
<Grid>
<!-- Content in Grid-->
<StackLayout
		touch:TouchEff.NativeAnimation="True"
		touch:TouchEff.NativeAnimationColor="Red"
		touch:TouchEff.Command="{Binding Command}"
		touch:TouchEff.CommandParameter="test">
<!--Content-->
</StackLayout>
</Grid>
<!-- Content in Grid-->
</ScrollView>
```
The problem is, if the user touches the StackLayout and begin to scroll the Animation effect begins. I only want the effect if the user "clicks" the StackLayout. Is this possible? 

Thank you.

**Comment by AndreiMisiukevich** [comment, marked as answer]
Hi, in this case, I propose to use regular TapGestureRecognizer (invoke animation on tap)

**Reply by AlleSchonWeg** [reply]
Hi @AndreiMisiukevich ,
i' not sure what you mean with "invoke animation on tap". Could you help me with a link?

Thanks again. 👍

**Reply by AndreiMisiukevich** [reply]
Well, if you want to see "Native" animation. It's not possible now.
But if you want to change color/opacity/scale, then just subsribe Tapped event of TapGestureRecognizer and perform desired animation e.g. 
```
view.FadeTo(0);
view.FadeTo(1);
```

**Reply by AlleSchonWeg** [reply]
Hi,
today i tried a few things, but nothing works well for me. The problem with the xct-toucheffekt is, that the effect begins at the down event. But the down event is executed if you touch and hold the view. For scrolling for example.

With CollectionView or ListView items the effect (Ripple on Android) starts on up event, if you not moving. I think i will create a CollectionView with only one element in ItemsSource. Hopefully this will work for me. 😀

Perhaps you accept feature request. ;)
It would be nice to have a property to start the animation at up event, if down event was executed and no scrolling was performed.

Thanks again!
